src/raster/rasterband.rs

```diff
@@ -125,6 +125,11 @@ impl <'a> RasterBand<'a> {
         None
     }
 
+    pub fn set_no_data_value(&self, no_data: f64) -> Result<()> {
+        unsafe { gdal_sys::GDALSetRasterNoDataValue(self.c_rasterband, no_data) };
```

Comment:
GDALSetRasterNoDataValue can return a CPL_Error. Please check if the call failed and if it does return an error.
I guess you can look at the ` pub fn write<T: GdalType+Copy>` method and copy the error handling from there :)

Reply:
Yup will do :+1: